docs/types/reference-types.rst

```diff
@@ -235,97 +235,69 @@ Array Literals
 ^^^^^^^^^^^^^^
 
 An array literal is a comma-separated list of one or more expressions, enclosed
-in square brackets (``[...]``). For example ``[1, a, f(3)]``. The type of the
-array literal is determined as follows:
-
-It is always a statically-sized memory array whose length is the
-number of expressions.
-
-The base type of the array is the type of the first expression on the list such that all
-other expressions can be implicitly converted to it. It is a type error
-if this is not possible.
-
-It is not enough that there is a type all the elements can be converted to. One of the elements
-has to be of that type.
-
-In the example below, the type of ``[1, 2, 3]`` is
-``uint8[3] memory``, because the type of each of these constants is ``uint8``. If
-you want the result to be a ``uint[3] memory`` type, you need to convert
-the first element to ``uint``.
+in square brackets (``[...]``). For example ``[1, a, f(3)]``. It can be converted to
+statically and dynamically-sized array if all its expressions can be implicitly
+converted to the base type of the array. In example below, the conversion is impossible
```

Comment:
done

Reply:
Yes, copying works with a way more than 16 elements. An interesting thing is the compiler seems to work way much faster and seamlessly without IR. Below you can find the results of tests. As a baseline I did tests for the solc 8.14. Optimization was enabled for all the tests.

The `Point` structure used in tests:
`struct Point {
        uint256 x;
        uint256 y;
    }`

via-IR:
| scenario | solc 8.14 | solc with changes |
|---|---|---|
|uint256[1000] memory a = [...];| 7m32s| 5m27s|
|Point[1000] memory a = [...];| 1m6s | exception |
|Point[240] memory a = [...];| NA | 1m38s|

no-IR:
| scenario | solc 8.14 | solc with changes |
|---|---|---|
|uint256[1000] memory a = [...];|4s| 2s|
|Point[1000] memory a = [...];| 1m6s | 2s |

The excpetion I get in one out of tests:
`...
Dynamic exception type: boost::wrapexcept<solidity::yul::YulAssertion>
std::exception::what: Could not create stack layout after 1000 iterations.
[solidity::util::tag_comment*] = Could not create stack layout after 1000 iterations.
`
I guess it requires attention.

Reply:
Haha, interesting :-)! Let me have a look - it may just be a matter of increasing the limit there.

This hints at a disadvantage of keeping things on stack, though: the EVM stack is only 1024 slots large in total... usually that's of little concern, but for such very large inline arrays, we may actually hit this limit, whereas we wouldn't have, if we kept constructing the array in memory right away... but maybe that's a reasonable limitation :-) - but something to think about, since the real-world array in ``MediumVerifier`` with 806 slots is not that far off that limit...

Reply:
Although, the Yul optimizer may actually transform the code into something that actually doesn't need a large number of stack slots, even if we first generate code like that...
In any case, it's generally a bit unfortunate that via-IR is extremely slow for these cases...